Crashes on cloudflare_notification_policy

### Confirmation

- [X] My issue isn't already found on the issue tracker.
- [X] I have replicated my issue using the latest version of the provider and it is still present.

### Terraform and Cloudflare provider version

Terraform v1.0.6
on linux_amd64
+ provider registry.terraform.io/cloudflare/cloudflare v2.26.1


### Affected resource(s)

cloudflare_notification_policy

### Terraform configuration files

```hcl
resource cloudflare_notification_policy notif {
	account_id = var.account_id
	name = "Pool enablement"
	description = "Pool enablement"
	enabled     =  true
	alert_type  = "g6_pool_toggle_alert"

	email_integration {
		id   =  "myemail@example.com"
	}
	filters = {
		enabled = ["true", "false"]
	}
}
```


### Debug output

[log.txt](https://github.com/cloudflare/terraform-provider-cloudflare/files/7152993/log.txt)


### Panic output

_No response_

### Expected output

I expect the ressource to be created without crashing terraform

### Actual output

Plugin crashes

### Steps to reproduce

1. Copy the configuration
2. Run terraform apply

### Additional factoids

_No response_

### References

_No response_

Still having this issue, more information about it (just for using the `filters` resource):

Terraform version: ``

<details>
```
Stack trace from the terraform-provider-cloudflare_v3.7.0 plugin:

panic: Unknown validation type: 6

goroutine 73 [running]:
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.validateMapValues({0xd1efdb, 0x7}, 0xc00064cfe0, 0xc000403b80, {0xc00064cf60, 0x1, 0x1})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/schema.go:1912 +0x969
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.schemaMap.validateMap(0xc27820, {0xd1efdb, 0x7}, {0xc27820, 0xc000638c00}, 0x8bbfac, 0xc0006432e8, {0xc00064cf60, 0x1, 0x1})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/schema.go:1819 +0x271
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.schemaMap.validateType(0xc00064cde0, {0xd1efdb, 0x7f2d6a}, {0xc27820, 0xc000638c00}, 0xc000403b80, 0xc158c0, {0xc00064cf60, 0x1, 0x1})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/schema.go:2111 +0x185
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.schemaMap.validate(0xc46d00, {0xd1efdb, 0x7}, 0xc000403b80, 0xc000643708, {0xc00064cf60, 0x1, 0x1})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/schema.go:1534 +0x71a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.schemaMap.validateObject(0xc000638c60, {0x0, 0xc000634210}, 0xcf7120, 0xc00064c8d0, {0x13ca548, 0xc27820, 0xc000638b40})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/schema.go:1970 +0x585
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.schemaMap.Validate(...)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/schema.go:649
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Validate(0xc0003ef260, 0xc0003d0300)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/resource.go:542 +0x4d
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Provider).ValidateResource(0xe4cd60, {0xc00052e140, 0x1e}, 0xc000638b40)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/provider.go:254 +0xf2
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ValidateResourceTypeConfig(0xc000320120, {0xc00063a1c0, 0xd1d11f}, 0xc0006341b0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/grpc_provider.go:230 +0x126
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ValidateResourceTypeConfig(0xc000322880, {0xe4c660, 0xc000638000}, 0xc00063a000)
	github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/tf5server/server.go:503 +0x2e2
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ValidateResourceTypeConfig_Handler({0xcede60, 0xc000322880}, {0xe4c660, 0xc000638000}, 0xc000636000, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:272 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0003388c0, {0xe5ae28, 0xc00049a000}, 0xc00053cc60, 0xc000451380, 0x138b2b0, 0x0)
	google.golang.org/grpc@v1.40.0/server.go:1297 +0xccf
google.golang.org/grpc.(*Server).handleStream(0xc0003388c0, {0xe5ae28, 0xc00049a000}, 0xc00053cc60, 0x0)
	google.golang.org/grpc@v1.40.0/server.go:1626 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.40.0/server.go:941 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.40.0/server.go:939 +0x294

Error: The terraform-provider-cloudflare_v3.7.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

╷
│ Error: Plugin did not respond
│ 
│   with cloudflare_zone.service,
│   on zones.tf line 17, in resource "cloudflare_zone" "service":
│   17: resource "cloudflare_zone" "service" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ValidateResourceConfig call. The plugin logs may
│ contain more details.
╵
ERRO[0031] 1 error occurred:
	* exit status 1

```
</details>

--- a/.changelog/1542.txt
+++ b/.changelog/1542.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+resource/cloudflare_notification_policy: Fix unexpected crashes when using cloudflare_notification_policy with a filters attribute
+```

--- a/cloudflare/resource_cloudflare_notification_policy.go
+++ b/cloudflare/resource_cloudflare_notification_policy.go
@@ -55,17 +55,21 @@ func resourceCloudflareNotificationPolicyRead(d *schema.ResourceData, meta inter
 	d.Set("enabled", policy.Result.Enabled)
 	d.Set("alert_type", policy.Result.AlertType)
 	d.Set("description", policy.Result.Description)
-	d.Set("filters", policy.Result.Filters)
-	d.Set("conditions", policy.Result.Conditions)
 	d.Set("created", policy.Result.Created.Format(time.RFC3339))
 	d.Set("modified", policy.Result.Modified.Format(time.RFC3339))
+
+	if err := d.Set("filters", flattenNotificationPolicyFilter(policy.Result.Filters)); err != nil {
+		return fmt.Errorf("failed to set filters: %s", err)
+	}
 
 	if err := d.Set("email_integration", setNotificationMechanisms(policy.Result.Mechanisms["email"])); err != nil {
 		return fmt.Errorf("failed to set email integration: %s", err)
 	}
+
 	if err := d.Set("pagerduty_integration", setNotificationMechanisms(policy.Result.Mechanisms["pagerduty"])); err != nil {
 		return fmt.Errorf("failed to set pagerduty integration: %s", err)
 	}
+
 	if err := d.Set("webhooks_integration", setNotificationMechanisms(policy.Result.Mechanisms["webhooks"])); err != nil {
 		return fmt.Errorf("failed to set webhooks integration: %s", err)
 	}
@@ -155,14 +159,36 @@ func buildNotificationPolicy(d *schema.ResourceData) cloudflare.NotificationPoli
 	}
 
 	if filters, ok := d.GetOk("filters"); ok {
-		notificationPolicy.Filters = filters.(map[string][]string)
-	}
-
-	if conditions, ok := d.GetOk("conditions"); ok {
-		notificationPolicy.Conditions = conditions.(map[string]interface{})
+		notificationPolicy.Filters = expandNotificationPolicyFilter(filters.([]interface{}))
 	}
 
 	return notificationPolicy
+}
+
+func expandNotificationPolicyFilter(list []interface{}) map[string][]string {
+	filters := make(map[string][]string)
+	for _, listItem := range list {
+		setMap := listItem.(map[string]interface{})
+		for k, mapItem := range setMap {
+			itemSet := mapItem.(*schema.Set)
+			for _, v := range itemSet.List() {
+				filters[k] = append(filters[k], v.(string))
+			}
+		}
+	}
+	return filters
+}
+
+func flattenNotificationPolicyFilter(filters map[string][]string) []interface{} {
+	filtersMap := make(map[string]interface{})
+	for k, v := range filters {
+		set := schema.NewSet(schema.HashString, []interface{}{})
+		for _, value := range v {
+			set.Add(value)
+		}
+		filtersMap[k] = set
+	}
+	return []interface{}{filtersMap}
 }
 
 func getNotificationMechanisms(s *schema.Set) []cloudflare.NotificationMechanismData {

--- a/cloudflare/resource_cloudflare_notification_policy.go
+++ b/cloudflare/resource_cloudflare_notification_policy.go
@@ -58,8 +58,10 @@ func resourceCloudflareNotificationPolicyRead(d *schema.ResourceData, meta inter
 	d.Set("created", policy.Result.Created.Format(time.RFC3339))
 	d.Set("modified", policy.Result.Modified.Format(time.RFC3339))
 
-	if err := d.Set("filters", flattenNotificationPolicyFilter(policy.Result.Filters)); err != nil {
-		return fmt.Errorf("failed to set filters: %s", err)
+	if policy.Result.Filters != nil && len(policy.Result.Filters) > 0 {
+		if err := d.Set("filters", flattenNotificationPolicyFilter(policy.Result.Filters)); err != nil {
+			return fmt.Errorf("failed to set filters: %s", err)
+		}
 	}
 
 	if err := d.Set("email_integration", setNotificationMechanisms(policy.Result.Mechanisms["email"])); err != nil {

--- a/cloudflare/resource_cloudflare_notification_policy.go
+++ b/cloudflare/resource_cloudflare_notification_policy.go
@@ -170,10 +170,8 @@ func buildNotificationPolicy(d *schema.ResourceData) cloudflare.NotificationPoli
 func expandNotificationPolicyFilter(list []interface{}) map[string][]string {
 	filters := make(map[string][]string)
 	for _, listItem := range list {
-		setMap := listItem.(map[string]interface{})
-		for k, mapItem := range setMap {
-			itemSet := mapItem.(*schema.Set)
-			for _, v := range itemSet.List() {
+		for k, mapItem := range listItem.(map[string]interface{}) {
+			for _, v := range mapItem.(*schema.Set).List() {
 				filters[k] = append(filters[k], v.(string))
 			}
 		}

--- a/cloudflare/resource_cloudflare_notification_policy_test.go
+++ b/cloudflare/resource_cloudflare_notification_policy_test.go
@@ -80,10 +80,9 @@ func testCheckCloudflareNotificationPolicyUpdated(resName, policyName, policyDes
 func TestAccCloudflareNotificationPolicy_WithFiltersAttribute(t *testing.T) {
 	rnd := generateRandomResourceName()
 	resourceName := "cloudflare_notification_policy." + rnd
-	updatedPolicyName := "durable objects usage notification"
+	updatedPolicyName := "updated workers usage notification"
 	updatedPolicyDesc := "updated description"
 	accountID := os.Getenv("CLOUDFLARE_ACCOUNT_ID")
-	zoneID := generateRandomResourceName()
 
 	resource.Test(t, resource.TestCase{
 		PreCheck: func() {
@@ -92,7 +91,7 @@ func TestAccCloudflareNotificationPolicy_WithFiltersAttribute(t *testing.T) {
 		Providers: testAccProviders,
 		Steps: []resource.TestStep{
 			{
-				Config: testCheckCloudflareNotificationPolicyWithFiltersAttribute(rnd, accountID, zoneID),
+				Config: testCheckCloudflareNotificationPolicyWithFiltersAttribute(rnd, accountID),
 				Check: resource.ComposeTestCheckFunc(
 					resource.TestCheckResourceAttr(resourceName, "name", "workers usage notification"),
 					resource.TestCheckResourceAttr(resourceName, "description", "test description"),
@@ -104,14 +103,14 @@ func TestAccCloudflareNotificationPolicy_WithFiltersAttribute(t *testing.T) {
 				),
 			},
 			{
-				Config: testCheckCloudflareNotificationPolicyWithFiltersAttributeUpdated(rnd, updatedPolicyName, updatedPolicyDesc, accountID, zoneID),
+				Config: testCheckCloudflareNotificationPolicyWithFiltersAttributeUpdated(rnd, updatedPolicyName, updatedPolicyDesc, accountID),
 				Check: resource.ComposeTestCheckFunc(
 					resource.TestCheckResourceAttr(resourceName, "name", updatedPolicyName),
 					resource.TestCheckResourceAttr(resourceName, "description", updatedPolicyDesc),
 					resource.TestCheckResourceAttr(resourceName, "enabled", "true"),
 					resource.TestCheckResourceAttr(resourceName, "alert_type", "billing_usage_alert"),
 					resource.TestCheckResourceAttr(resourceName, "account_id", accountID),
-					resource.TestCheckTypeSetElemAttr(resourceName, "filters.0.product.*", "worker_durable_objects_requests"),
+					resource.TestCheckTypeSetElemAttr(resourceName, "filters.0.product.*", "worker_requests"),
 					resource.TestCheckTypeSetElemAttr(resourceName, "filters.0.limit.*", "100"),
 				),
 			},
@@ -119,7 +118,7 @@ func TestAccCloudflareNotificationPolicy_WithFiltersAttribute(t *testing.T) {
 	})
 }
 
-func testCheckCloudflareNotificationPolicyWithFiltersAttribute(name, accountID, zoneID string) string {
+func testCheckCloudflareNotificationPolicyWithFiltersAttribute(name, accountID string) string {
 	return fmt.Sprintf(`
   resource "cloudflare_notification_policy" "%[1]s" {
     name        = "workers usage notification"
@@ -137,10 +136,10 @@ func testCheckCloudflareNotificationPolicyWithFiltersAttribute(name, accountID, 
       ]
 	  limit = ["100"]
 	}
-  }`, name, accountID, zoneID)
+  }`, name, accountID)
 }
 
-func testCheckCloudflareNotificationPolicyWithFiltersAttributeUpdated(name, policyName, policyDesc, accountID, zoneID string) string {
+func testCheckCloudflareNotificationPolicyWithFiltersAttributeUpdated(name, policyName, policyDesc, accountID string) string {
 	return fmt.Sprintf(`
   resource "cloudflare_notification_policy" "%[1]s" {
     name        = "%[2]s"
@@ -154,11 +153,11 @@ func testCheckCloudflareNotificationPolicyWithFiltersAttributeUpdated(name, poli
     }
     filters {
       product = [
-        "worker_durable_objects_requests",
+        "worker_requests",
       ]
       limit = ["100"]
 	}
-  }`, name, policyName, policyDesc, accountID, zoneID)
+  }`, name, policyName, policyDesc, accountID)
 }
 
 func TestFlattenExpandFilters(t *testing.T) {

--- a/cloudflare/resource_cloudflare_notification_policy_test.go
+++ b/cloudflare/resource_cloudflare_notification_policy_test.go
@@ -11,13 +11,6 @@ import (
 )
 
 func TestAccCloudflareNotificationPolicy_Basic(t *testing.T) {
-	if os.Getenv("CLOUDFLARE_API_TOKEN") != "" {
-		defer func(apiToken string) {
-			os.Setenv("CLOUDFLARE_API_TOKEN", apiToken)
-		}(os.Getenv("CLOUDFLARE_API_TOKEN"))
-		os.Setenv("CLOUDFLARE_API_TOKEN", "")
-	}
-
 	rnd := generateRandomResourceName()
 	resourceName := "cloudflare_notification_policy." + rnd
 	updatedPolicyName := "updated test SSL policy from terraform provider"

--- a/cloudflare/resource_cloudflare_notification_policy_test.go
+++ b/cloudflare/resource_cloudflare_notification_policy_test.go
@@ -2,16 +2,15 @@ package cloudflare
 
 import (
 	"fmt"
+	"github.com/stretchr/testify/assert"
 	"os"
+	"sort"
 	"testing"
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 )
 
-func TestAccCloudflareNotificationPolicy(t *testing.T) {
-	// Temporarily unset CLOUDFLARE_API_TOKEN if it is set as the notification
-	// service does not yet support the API tokens and it results in
-	// misleading state error messages.
+func TestAccCloudflareNotificationPolicy_Basic(t *testing.T) {
 	if os.Getenv("CLOUDFLARE_API_TOKEN") != "" {
 		defer func(apiToken string) {
 			os.Setenv("CLOUDFLARE_API_TOKEN", apiToken)
@@ -83,4 +82,102 @@ func testCheckCloudflareNotificationPolicyUpdated(resName, policyName, policyDes
       id   =  "test@example.com"
     }
   }`, resName, policyName, policyDesc, accountID)
+}
+
+func TestAccCloudflareNotificationPolicy_WithFiltersAttribute(t *testing.T) {
+	rnd := generateRandomResourceName()
+	resourceName := "cloudflare_notification_policy." + rnd
+	updatedPolicyName := "durable objects usage notification"
+	updatedPolicyDesc := "updated description"
+	accountID := os.Getenv("CLOUDFLARE_ACCOUNT_ID")
+	zoneID := generateRandomResourceName()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheckAccount(t)
+		},
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testCheckCloudflareNotificationPolicyWithFiltersAttribute(rnd, accountID, zoneID),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "name", "workers usage notification"),
+					resource.TestCheckResourceAttr(resourceName, "description", "test description"),
+					resource.TestCheckResourceAttr(resourceName, "enabled", "true"),
+					resource.TestCheckResourceAttr(resourceName, "alert_type", "billing_usage_alert"),
+					resource.TestCheckResourceAttr(resourceName, "account_id", accountID),
+					resource.TestCheckTypeSetElemAttr(resourceName, "filters.0.product.*", "worker_requests"),
+					resource.TestCheckTypeSetElemAttr(resourceName, "filters.0.limit.*", "100"),
+				),
+			},
+			{
+				Config: testCheckCloudflareNotificationPolicyWithFiltersAttributeUpdated(rnd, updatedPolicyName, updatedPolicyDesc, accountID, zoneID),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "name", updatedPolicyName),
+					resource.TestCheckResourceAttr(resourceName, "description", updatedPolicyDesc),
+					resource.TestCheckResourceAttr(resourceName, "enabled", "true"),
+					resource.TestCheckResourceAttr(resourceName, "alert_type", "billing_usage_alert"),
+					resource.TestCheckResourceAttr(resourceName, "account_id", accountID),
+					resource.TestCheckTypeSetElemAttr(resourceName, "filters.0.product.*", "worker_durable_objects_requests"),
+					resource.TestCheckTypeSetElemAttr(resourceName, "filters.0.limit.*", "100"),
+				),
+			},
+		},
+	})
+}
+
+func testCheckCloudflareNotificationPolicyWithFiltersAttribute(name, accountID, zoneID string) string {
+	return fmt.Sprintf(`
+  resource "cloudflare_notification_policy" "%[1]s" {
+    name        = "workers usage notification"
+    account_id  = "%[2]s"
+    description = "test description"
+    enabled     =  true
+    alert_type  = "billing_usage_alert"
+    email_integration {
+      name =  ""
+      id   =  "test@example.com"
+    }
+    filters {
+      product = [
+        "worker_requests",
+      ]
+	  limit = ["100"]
+	}
+  }`, name, accountID, zoneID)
+}
+
+func testCheckCloudflareNotificationPolicyWithFiltersAttributeUpdated(name, policyName, policyDesc, accountID, zoneID string) string {
+	return fmt.Sprintf(`
+  resource "cloudflare_notification_policy" "%[1]s" {
+    name        = "%[2]s"
+    account_id  = "%[4]s"
+    description = "%[3]s"
+    enabled     =  true
+    alert_type  = "billing_usage_alert"
+    email_integration {
+      name =  ""
+      id   =  "test@example.com"
+    }
+    filters {
+      product = [
+        "worker_durable_objects_requests",
+      ]
+      limit = ["100"]
+	}
+  }`, name, policyName, policyDesc, accountID, zoneID)
+}
+
+func TestFlattenExpandFilters(t *testing.T) {
+	filters := map[string][]string{
+		"services": []string{"waf", "firewallrules"},
+		"zones":    []string{"abc123"},
+	}
+	flattenedFilters := flattenNotificationPolicyFilter(filters)
+	expandedFilters := expandNotificationPolicyFilter(flattenedFilters)
+	for k, _ := range filters {
+		sort.Strings(filters[k])
+		sort.Strings(expandedFilters[k])
+		assert.EqualValuesf(t, filters[k], expandedFilters[k], "values should equal without order")
+	}
 }

--- a/cloudflare/schema_cloudflare_notification_policy.go
+++ b/cloudflare/schema_cloudflare_notification_policy.go
@@ -33,10 +33,6 @@ func resourceCloudflareNotificationPolicySchema() map[string]*schema.Schema {
 			Type:     schema.TypeString,
 			Computed: true,
 		},
-		//"conditions": {
-		//	Type:     schema.TypeMap,
-		//	Optional: true,
-		//},
 		"email_integration": {
 			Type:     schema.TypeSet,
 			Optional: true,
@@ -81,9 +77,10 @@ func notificationPolicyFilterSchema() *schema.Schema {
 					Optional: true,
 				},
 				"health_check_id": {
-					Type:     schema.TypeSet,
-					Elem:     &schema.Schema{Type: schema.TypeString},
-					Optional: true,
+					Type:         schema.TypeSet,
+					Elem:         &schema.Schema{Type: schema.TypeString},
+					Optional:     true,
+					RequiredWith: []string{"filters.0.status"},
 				},
 				"zones": {
 					Type: schema.TypeSet,

--- a/cloudflare/schema_cloudflare_notification_policy.go
+++ b/cloudflare/schema_cloudflare_notification_policy.go
@@ -110,6 +110,21 @@ func notificationPolicyFilterSchema() *schema.Schema {
 					},
 					Optional: true,
 				},
+				"enabled": {
+					Type: schema.TypeSet,
+					Elem: &schema.Schema{
+						Type: schema.TypeString,
+					},
+					Optional: true,
+				},
+				"pool_id": {
+					Type: schema.TypeSet,
+					Elem: &schema.Schema{
+						Type: schema.TypeString,
+					},
+					Optional:     true,
+					RequiredWith: []string{"filters.0.enabled"},
+				},
 			},
 		},
 	}

--- a/cloudflare/schema_cloudflare_notification_policy.go
+++ b/cloudflare/schema_cloudflare_notification_policy.go
@@ -24,14 +24,7 @@ func resourceCloudflareNotificationPolicySchema() map[string]*schema.Schema {
 			Type:     schema.TypeString,
 			Required: true,
 		},
-		"filters": {
-			Type:     schema.TypeMap,
-			Optional: true,
-			Elem: &schema.Schema{
-				Type: schema.TypeList,
-				Elem: schema.TypeString,
-			},
-		},
+		"filters": notificationPolicyFilterSchema(),
 		"created": {
 			Type:     schema.TypeString,
 			Computed: true,
@@ -40,10 +33,10 @@ func resourceCloudflareNotificationPolicySchema() map[string]*schema.Schema {
 			Type:     schema.TypeString,
 			Computed: true,
 		},
-		"conditions": {
-			Type:     schema.TypeMap,
-			Optional: true,
-		},
+		//"conditions": {
+		//	Type:     schema.TypeMap,
+		//	Optional: true,
+		//},
 		"email_integration": {
 			Type:     schema.TypeSet,
 			Optional: true,
@@ -73,4 +66,54 @@ var mechanismData = &schema.Resource{
 			Optional: true,
 		},
 	},
+}
+
+func notificationPolicyFilterSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		MaxItems: 1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"status": {
+					Type:     schema.TypeSet,
+					Elem:     &schema.Schema{Type: schema.TypeString},
+					Optional: true,
+				},
+				"health_check_id": {
+					Type:     schema.TypeSet,
+					Elem:     &schema.Schema{Type: schema.TypeString},
+					Optional: true,
+				},
+				"zones": {
+					Type: schema.TypeSet,
+					Elem: &schema.Schema{
+						Type: schema.TypeString,
+					},
+					Optional: true,
+				},
+				"services": {
+					Type: schema.TypeSet,
+					Elem: &schema.Schema{
+						Type: schema.TypeString,
+					},
+					Optional: true,
+				},
+				"product": {
+					Type: schema.TypeSet,
+					Elem: &schema.Schema{
+						Type: schema.TypeString,
+					},
+					Optional: true,
+				},
+				"limit": {
+					Type: schema.TypeSet,
+					Elem: &schema.Schema{
+						Type: schema.TypeString,
+					},
+					Optional: true,
+				},
+			},
+		},
+	}
 }

--- a/website/docs/r/notification_policy.html.markdown
+++ b/website/docs/r/notification_policy.html.markdown
@@ -14,6 +14,7 @@ mechanisms supported are email, webhooks, and PagerDuty.
 ## Example Usage
 
 ### Basic Example
+
 ```hcl
 resource "cloudflare_notification_policy" "example" {
   account_id = "c4a7362d577a6c3019a474fd6f485821"
@@ -36,7 +37,8 @@ resource "cloudflare_notification_policy" "example" {
 }
 ```
 
-Example With Filters
+### With Filters
+
 ```hcl
 resource "cloudflare_notification_policy" "example" {
   account_id  = "c4a7362d577a6c3019a474fd6f485821"
@@ -56,7 +58,7 @@ resource "cloudflare_notification_policy" "example" {
   pagerduty_integration {
     id   =  "850129d136459401860572c5d964d27k"
   }
-  
+
   filters {
     health_check_id = ["699d98642c564d2e855e9661899b7252"]
     status           = ["Unhealthy"]
@@ -71,29 +73,49 @@ The following arguments are supported:
 * `account_id` - (Required) The ID of the account for which the notification policy has to be created.
 * `name` - (Required) The name of the notification policy.
 * `enabled` - (Required) The status of the notification policy, a boolean value.
-* `alert_type` - (Required) The event type that will trigger the dispatch of a notification.
+* `alert_type` - (Required) The event type that will trigger the dispatch of a notification (refer to the [nested schema](#nestedblock--alert-type)).
 * `email_integration` - (Optional) The email id to which the notification should be dispatched. One of email, webhooks, or PagerDuty mechanisms is required.
 * `webhooks_integration` - (Optional) The unique id of a configured webhooks endpoint to which the notification should be dispatched. One of email, webhooks, or PagerDuty mechanisms is required.
 * `pagerduty_integration` - (Optional) The unique id of a configured pagerduty endpoint to which the notification should be dispatched. One of email, webhooks, or PagerDuty mechanisms is required.
 * `description` - (Optional) Description of the notification policy.
-* `filters` - (Optional) An optional nested block of filters that applies to the selected `alert_type`. A key-value map that specifies the type of filter and the values to match against.
+* `filters` - (Optional) An optional nested block of filters that applies to the selected `alert_type`. A key-value map that specifies the type of filter and the values to match against (refer to the alert type block for available fields).
 
-## Filters
+<a id="nestedblock--alert-type"></a>
+**Nested schema for `alert_type`**
 
-| Alert Type                       | Filter          | Description                               | Example Values                                                                                                                                                                                                                                                                                    |
-|----------------------------------|-----------------|-------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
-| billing_usage_alert              |                 | billing usage exceeds threshold           |                                                                                                                                                                                                                                                                                                   |
-|                                  | product         | product name                              | "worker_requests", "worker_durable_objects_requests", "worker_durable_objects_duration", "worker_durable_objects_data_transfer", "worker_durable_objects_stored_data", "worker_durable_objects_storage_deletes", "worker_durable_objects_storage_writes", "worker_durable_objects_storage_reads"  |
-|                                  | limit           | a numerical limit                         | "100"                                                                                                                                                                                                                                                                                             |
-| health_check_status_notification |                 | health check status changes               |                                                                                                                                                                                                                                                                                                   |
-|                                  | health_check_id | health check ID                           | 699d98642c564d2e855e9661899b7252                                                                                                                                                                                                                                                                  |
-|                                  | status          | status to alert on                        | "Unhealthy", "Healthy"                                                                                                                                                                                                                                                                            |
-| g6_pool_toggle_alert             |                 | pool alerts on enable/disable status      |                                                                                                                                                                                                                                                                                                   |
-|                                  | pool_id         | load balancing pool id                    | "17b5962d775c646f3f9725cbc7a53df4"                                                                                                                                                                                                                                                                |
-|                                  | enabled         | state to alert on                         | "true", "false"                                                                                                                                                                                                                                                                                   |
-| real_origin_monitoring           |                 | Cloudflare is unable to reach your origin |                                                                                                                                                                                                                                                                                                   |
-| universal_ssl_event_type         |                 | universal certificate notices             |                                                                                                                                                                                                                                                                                                   |
-| bgp_hijack_notification          |                 | alerts for BGP hijack                     |                                                                                                                                                                                                                                                                                                   |
+* `billing_usage_alert` - (Optional) Billing usage exceeds threshold (refer to the [nested schema](#nestedblock--alert-type-billing-usage-alert)).
+* `health_check_status_notification` - (Optional) Health check status changes (refer to the [nested schema](#nestedblock--alert-type-health-check-status-notification)).
+* `g6_pool_toggle_alert` - (Optional) Pool alerts on enable/disable status (refer to the [nested schema](#nestedblock--alert-type-g6-pool-toggle-alert)).
+* `real_origin_monitoring` - (Optional) Cloudflare is unable to reach your origin (refer to the [nested schema](#nestedblock--alert-type-real-origin-monitoring)).
+* `universal_ssl_event_type` - (Optional) Universal certificate notices (refer to the [nested schema](#nestedblock--alert-type-universal-ssl-event-type)).
+* `bgp_hijack_notification` - (Optional) Alerts for BGP hijack (refer to the [nested schema](#nestedblock--alert-type-bgp-hijack-notification)).
+
+<a id="nestedblock--alert-type-billing-usage-alert"></a>
+**Nested schema for `billing_usage_alert`**
+
+* `product` - (Optional) Product name. Available values: `"worker_requests"`, `"worker_durable_objects_requests"`, `"worker_durable_objects_duration"`, `"worker_durable_objects_data_transfer"`, `"worker_durable_objects_stored_data"`, `"worker_durable_objects_storage_deletes"`, `"worker_durable_objects_storage_writes"`, `"worker_durable_objects_storage_reads"`.
+* `limit` - (Optional) A numerical limit. Example: `"100"`
+
+<a id="nestedblock--alert-type-health-check-status-notification"></a>
+**Nested schema for `health_check_status_notification`**
+
+* `health_check_id` - (Optional) Identifier health check.
+* `status` - (Optional) Status to alert on. Example: `"Unhealthy"`, `"Healthy"`.
+
+<a id="nestedblock--alert-type-g6-pool-toggle-alert"></a>
+**Nested schema for `g6_pool_toggle_alert`**
+
+* `pool_id` - (Optional) Load balancer pool identifier.
+* `enabled` - (Optional) State of the pool to alert on. Example: `"true"`, `"false"`.
+
+<a id="#nestedblock--alert-type-real-origin-monitoring"></a>
+**Nested schema for `real_origin_monitoring`**
+
+<a id="#nestedblock--alert-type-universal-ssl-event-type"></a>
+**Nested schema for `universal_ssl_event_type`**
+
+<a id="#nestedblock--alert-type-bgp-hijack-notification"></a>
+**Nested schema for `bgp_hijack_notification`**
 
 ## Import
 

--- a/website/docs/r/notification_policy.html.markdown
+++ b/website/docs/r/notification_policy.html.markdown
@@ -13,6 +13,7 @@ mechanisms supported are email, webhooks, and PagerDuty.
 
 ## Example Usage
 
+### Basic Example
 ```hcl
 resource "cloudflare_notification_policy" "example" {
   account_id = "c4a7362d577a6c3019a474fd6f485821"
@@ -35,6 +36,34 @@ resource "cloudflare_notification_policy" "example" {
 }
 ```
 
+Example With Filters
+```hcl
+resource "cloudflare_notification_policy" "example" {
+  account_id  = "c4a7362d577a6c3019a474fd6f485821"
+  name        = "Policy for Healthcheck notification"
+  description = "Notification policy to alert on unhealthy Healthcheck status"
+  enabled     =  true
+  alert_type  = "health_check_status_notification"
+
+  email_integration {
+    id   =  "myemail@example.com"
+  }
+
+  webhooks_integration {
+    id   =  "1860572c5d964d27aa0f379d13645940"
+  }
+
+  pagerduty_integration {
+    id   =  "850129d136459401860572c5d964d27k"
+  }
+  
+  filters {
+    health_check_id = ["699d98642c564d2e855e9661899b7252"]
+    status           = ["Unhealthy"]
+  }
+}
+```
+
 ## Argument Reference
 
 The following arguments are supported:
@@ -47,7 +76,24 @@ The following arguments are supported:
 * `webhooks_integration` - (Optional) The unique id of a configured webhooks endpoint to which the notification should be dispatched. One of email, webhooks, or PagerDuty mechanisms is required.
 * `pagerduty_integration` - (Optional) The unique id of a configured pagerduty endpoint to which the notification should be dispatched. One of email, webhooks, or PagerDuty mechanisms is required.
 * `description` - (Optional) Description of the notification policy.
-* `filters` - (Optional) Optional filterable items for a policy.
+* `filters` - (Optional) An optional nested block of filters that applies to the selected `alert_type`. A key-value map that specifies the type of filter and the values to match against.
+
+## Filters
+
+| Alert Type                       | Filter          | Description                               | Example Values                                                                                                                                                                                                                                                                                    |
+|----------------------------------|-----------------|-------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| billing_usage_alert              |                 | billing usage exceeds threshold           |                                                                                                                                                                                                                                                                                                   |
+|                                  | product         | product name                              | "worker_requests", "worker_durable_objects_requests", "worker_durable_objects_duration", "worker_durable_objects_data_transfer", "worker_durable_objects_stored_data", "worker_durable_objects_storage_deletes", "worker_durable_objects_storage_writes", "worker_durable_objects_storage_reads"  |
+|                                  | limit           | a numerical limit                         | "100"                                                                                                                                                                                                                                                                                             |
+| health_check_status_notification |                 | health check status changes               |                                                                                                                                                                                                                                                                                                   |
+|                                  | health_check_id | health check ID                           | 699d98642c564d2e855e9661899b7252                                                                                                                                                                                                                                                                  |
+|                                  | status          | status to alert on                        | "Unhealthy", "Healthy"                                                                                                                                                                                                                                                                            |
+| g6_pool_toggle_alert             |                 | pool alerts on enable/disable status      |                                                                                                                                                                                                                                                                                                   |
+|                                  | pool_id         | load balancing pool id                    | "17b5962d775c646f3f9725cbc7a53df4"                                                                                                                                                                                                                                                                |
+|                                  | enabled         | state to alert on                         | "true", "false"                                                                                                                                                                                                                                                                                   |
+| real_origin_monitoring           |                 | Cloudflare is unable to reach your origin |                                                                                                                                                                                                                                                                                                   |
+| universal_ssl_event_type         |                 | universal certificate notices             |                                                                                                                                                                                                                                                                                                   |
+| bgp_hijack_notification          |                 | alerts for BGP hijack                     |                                                                                                                                                                                                                                                                                                   |
 
 ## Import
 

